inc/class-tmsc-sync.php

```diff
@@ -356,43 +385,84 @@ public function do_post_processing() {
 											'meta_query' => array(
 												array(
 													'key'     => 'tmsc_legacy_id',
-													'value'   => $constituent_ids,
+													'value'   => $ids,
 													'compare' => 'IN',
 												),
 											),
 										) );
-										if ( ! empty( $related_constituents ) ) {
-											$meta_data[ $role_slug ] = $related_constituents;
+
+										if ( ! empty( $related_posts ) && $existing_relationships !== $related_posts ) {
+											update_post_meta( $post_id, $key, $related_posts );
+										}
+									} elseif ( 'constituent' === $relationship_map[ $key ]['type'] ) {
+										foreach ( $ids as $constituent_type_slug => $role_data ) {
+											$meta_data = array();
+											foreach( $role_data as $role_slug => $constituent_ids ) {
```

Comment:
Do you need to ensure ` ! empty( $role_data ) && is_array( $role_data )` before looping through `$role_data`?